src/main/java/Duke.java

```diff
@@ -1,10 +1,275 @@
+
+        // get the date and time of the deadline in the format of dd/mm/yyyy HHmm
+        String by = userInput.replaceAll(RegexType.GET_BY_REGEX.getRegexType(), "");
+
+        // Create new Deadline instance an add it to end taskList
+        taskList.add(new Deadline (description, by));
+
+        printAcknowledgement(RegexType.TASK_ADDED_LINE.getRegexType(), taskList.size()-1);
+        printNumOfTask();
+    }
+
+    // Adding an Event to list of tasks
+    public static void addEvent(String userInput) throws EventException{
+        // Check if userInput follows the format "<description> /at <time/date>"
+        //Make new regex for event which shall understand the pattern of date and time as per level 8 - as of now this has been resolved
+        if (!userInput.matches(RegexType.EVENT_REGEX.getRegexType())){
+            throw new EventException();
+        }
+
+        // get description and by from userInput
+        String description = userInput.replaceAll(RegexType.GET_DESCRIPTION_REGEX.getRegexType(), "");
+        String at = userInput.replaceAll(RegexType.GET_AT_REGEX.getRegexType(), "");
+
+        // Create new Event instance an add it to end taskList
+        taskList.add(new Event (description, at));
+
+        printAcknowledgement(RegexType.TASK_ADDED_LINE.getRegexType(), taskList.size()-1);
+        printNumOfTask();
+    }
+
+    public static void addNothing() throws NothingException{
+        throw new NothingException();
+    }
+
+    // Delete line
+    public static void deleteTask(String userInput)
+            throws DeleteFormatException, DeleteRangeException {
+
+        // Check if the command is done and is followed by a number
+        // and if the index is within the range of number of tasks
+        if (!userInput.matches(RegexType.DIGITS_REGEX.getRegexType())){
+            throw new DeleteFormatException();
+        }
+
+        int index = Integer.parseInt(userInput) - 1;
+        if (index >= taskList.size()){
+            throw new DeleteRangeException();
+        }
+
+        printAcknowledgement(PrintType.TASK_DELETED_LINE.getPrintType(), index);
+
+        taskList.remove(index);
+
+        printNumOfTask();
+    }
+
+    // Mark the task at the given index as done
+    public static void markAsDone(String userInput)
+            throws DoneFormatException, DoneAlreadyException, DoneRangeException{
+
+        // Check if the command is done and is followed by a number
+        // and if the index is within the range of number of tasks
+        if (!userInput.matches(RegexType.DIGITS_REGEX.getRegexType())){
+            throw new DoneFormatException();
+        }
+
+        int index = Integer.parseInt(userInput) - 1;
+        if (index >= taskList.size()){
+            throw new DoneRangeException();
+        }
+        // Check if task is already done
+        if (taskList.get(index).isDone()){
+            throw new DoneAlreadyException();
+        }
+
+        // Mark the index as done
```

Comment:
Perhaps proper javadocs could be added so IDE can display the documentation with all the information?